pkg/tfbridge/info.go

```diff
@@ -180,6 +180,9 @@ type ResourceInfo struct {
 	DeprecationMessage  string      // message to use in deprecation warning
 	CSharpName          string      // .NET-specific name
 
+	// Optional hook to run before upgrading the state. This is currently only supported for Plugin-Framework based
+	// providers.
+	PreStateUpgradeHook PreStateUpgradeHook
```

Comment:
Do we plan to support `PreStateUpgradeHook` on all bridged providers in the future? If so, can you comment with the tracking issue. If not, could we put this in `pf/tfbridge.ProviderInfo`?
https://github.com/pulumi/pulumi-terraform-bridge/blob/92f231195177017c423321aa4f0c17d524f17e1d/pf/tfbridge/info.go#L22-L33

Reply:
Pondering this, a bit of a design question.. I think it should be implementable and occasionally useful for pre-PF providers as well. I've added an issue:

https://github.com/pulumi/pulumi-terraform-bridge/issues/864